Client/Frontend/Browser/Playlist/Controllers/PlaylistCarplayController.swift

```diff
@@ -63,38 +62,49 @@ class PlaylistCarplayController: NSObject {
     }
     
     func observeFolderStates() {
-        folderObserver = PlaylistManager.shared.onCurrentFolderDidChange
-            .receive(on: DispatchQueue.main)
-            .sink { [weak self] in
-                guard let self = self else { return }
-                
-                if self.interfaceController.rootTemplate == self.interfaceController.topTemplate {
-                    // Nothing to Pop
+        PlaylistCarplayManager.shared.onCarplayUIChangedToRoot.eraseToAnyPublisher()
```

Comment:
`onCurrentFolderDidChange` doesnt also satisfy the condition for `onCarplayUIChangedToRoot` ? why do we need this separately?

Reply:
Because folder change doesn't necessarily equate to UI change. I can go back to the `root` folders view which changes the UI, but I don't have to select a folder, and so one will not fire.